src/modules/navigator/geofence.cpp

```diff
@@ -118,13 +132,50 @@ bool Geofence::inside(const struct vehicle_global_position_s &global_position,
 
 bool Geofence::inside(double lat, double lon, float altitude)
 {
+	if (_param_geofence_mode.get() >= GEOFENCE_MAX_DISTANCES_ONLY) {
+		int32_t max_horizontal_distance = _param_max_hor_distance.get();
+		int32_t max_vertical_distance = _param_max_ver_distance.get();
+
+		if (max_horizontal_distance > 0 || max_vertical_distance > 0) {
+			bool updated = false;
+			orb_check(_home_position_sub, &updated);
+
+			if (updated) {
+				/* get a local copy of the home position data */
+				struct home_position_s home;
+				memset(&home, 0, sizeof(home));
+				orb_copy(ORB_ID(home_position), _home_position_sub, &home);
+
+				_home_position_set = true;
+
+				_home_lat = home.lat;
+				_home_lon = home.lon;
+				_home_alt = home.alt;
+			}
+
+			if (_home_position_set) {
+				struct home_position_s home;
+				orb_copy(ORB_ID(home_position), _home_position_sub, &home);
+
+				float dist_xy = -1.0f;
+				float dist_z = -1.0f;
+				get_distance_to_point_global_wgs84(lat, lon, altitude, _home_lat, _home_lon, _home_alt, &dist_xy, &dist_z);
+
+				if ((max_vertical_distance > 0 && (dist_z > max_vertical_distance))
+				    || (max_horizontal_distance > 0 && (dist_xy > max_horizontal_distance))) {
+					return false;
+				}
+			}
+		}
+	}
+
 	bool inside_fence = inside_polygon(lat, lon, altitude);
```

Comment:
Shouldn't there be a second check for param_geofence_mode somewhere here? eg run inside_polygon depending on that parameter?